poppy/poppy_core.py

```diff
@@ -687,16 +699,31 @@ def _resample_wavefront_pixelscale(self, detector):
         _log.debug("Desired detector FOV: {} pixels, {:.3f}".format(detector.shape,
                                                                     detector.shape[0]*u.pixel*detector.pixelscale))
 
-        # TODO the following works but is not optimal for performance
-        # We should consider cropping out an appropriate subregion prior to performing the zoom.
-        # That makes a difference if the detector is only sampling a small part of a much larger wavefront
-
-        new_wf_real = scipy.ndimage.zoom(self.wavefront.real, pixscale_ratio, order=detector.interp_order)
-        new_wf_imag = scipy.ndimage.zoom(self.wavefront.imag, pixscale_ratio, order=detector.interp_order)
-        new_wf = new_wf_real + 1.j*new_wf_imag
+        def make_axis(npix, step):
+            """ Helper function to make coordinate axis for interpolation """
+            return step * np.arange(-npix // 2, npix // 2, dtype=np.float64)
+
+        # Provide 2x margin around image to reduce interpolation errors at edge, but also make
+        # sure that image is centered properly after it gets cropped down to detector size
+        margin_factor = 2
+        x_in = make_axis(self.wavefront.shape[0], self.pixelscale)
+        y_in = make_axis(self.wavefront.shape[1], self.pixelscale)
+        x_out = make_axis(margin_factor * detector.shape[0], detector.pixelscale)
+        y_out = make_axis(margin_factor * detector.shape[1], detector.pixelscale)
+
+        def interpolator(arr):
+            """ Bind arguments to scipy's interp2d function """
+            # TODO: scipy.interpolate.RectBivariateSpline is much more efficient for data on a
```

Comment:
Would it make sense to try to address this `TODO` idea prior to merging this in?